testsuites/syncgateway/functional/tests/test_webhooks.py

```diff
@@ -310,6 +314,237 @@ def test_webhooks_crud(params_from_base_test_setup, sg_conf_name, filtered):
+    js_func_key = "\"filter\":\""
+    path = "https://{}:5007/webhookFilter".format(get_local_ip())
+    path = js_func_key + path + "\","
+    temp_sg_config, _ = copy_sgconf_to_temp(sg_conf, mode)
+    temp_sg_config = replace_string_on_sgw_config(temp_sg_config, "{{ webhook_filter }}", path)
+    cluster.reset(sg_config_path=temp_sg_config)
+    webhook_server.start()
+
+    topology = cluster_helper.get_cluster_topology(cluster_config)
+
+    cbs_url = topology["couchbase_servers"][0]
+    sg_url = topology["sync_gateways"][0]["public"]
+    sg_url_admin = topology["sync_gateways"][0]["admin"]
+    sg_db = "db"
+
+    log_info("Running 'test_attachment_revpos_when_ancestor_unavailable'")
+    log_info("Using cbs_url: {}".format(cbs_url))
+    log_info("Using sg_url: {}".format(sg_url))
+    log_info("Using sg_url_admin: {}".format(sg_url_admin))
+    log_info("Using sg_db: {}".format(sg_db))
+    log_info("Using bucket: {}".format(bucket))
+
+    sg_client.create_user(sg_admin_url, sg_db, username, password=password, channels=channel)
+    cookie, session_id = sg_client.create_session(sg_admin_url, sg_db, username)
+    # user_session = cookie, session_id
+
+    # webhook_filter verification
+    def update_webhook_prop():
+        return {'updates': 0, 'data': 'webhook_filter'}
+
+    def update_non_webhook_prop():
+        return {'updates': 0, 'data': 'non_webhook_filter'}
+    log_info('Adding {} docs via SDK ...')
+    sdk_doc_bodies = document.create_docs(sdk_webhook, number=sdk_webhook_docs, content={"data": "webhook_filter"}, channels=channel, prop_generator=update_webhook_prop)
+    sdk_doc_ids1 = [doc['_id'] for doc in sdk_doc_bodies]
+    sdk_docs = {doc['_id']: doc for doc in sdk_doc_bodies}
+    sdk_client.upsert_multi(sdk_docs)
+
+    sdk_doc_bodies = document.create_docs(sdk_non_webhook, number=sdk_non_webhook_docs, content={"data": "non_webhook_filter"}, channels=channel)
+    sdk_docs = {doc['_id']: doc for doc in sdk_doc_bodies}
+    sdk_client.upsert_multi(sdk_docs)
+
+    count = 0
+    retries = 10
+    while count < retries:
+        data = webhook_server.get_data()
+        # Remove unwanted data from the response
+        for item in data:
+            if "_id" not in item:
+                data.remove(item)
+
+        posted_webhook_events_ids = [item['_id'] for item in data]
+        if len(posted_webhook_events_ids) < len(sdk_doc_ids1):
+            time.sleep(2)
+            count += 1
+            continue
+        else:
+            break
+    assert len(posted_webhook_events_ids) == len(sdk_doc_ids1)
+
+
+@pytest.fixture(scope="function")
+def setup_webserver():
+    webhook_server = WebServer()
+    process = subprocess.Popen(args=["nohup", "python", "libraries/utilities/host_sgw_jscode.py", "--start", "&"], stdout=subprocess.PIPE)
+    yield{
+        "webhook_server": webhook_server
+    }
+
+    webhook_server.stop()
+    process.kill()
+
+
+@pytest.fixture(scope="function")
+def setup_webserver_jssslon():
```

Comment:
i am kind of guessing what this function name means, if you don't mind, setup_webserver_js_ssl_on() may be more straightforward. :)